scripts/update_rounds_info.py

```diff
@@ -450,6 +448,7 @@ def write_updated_rounds_info(updated_rounds_info: Dict[str, RoundInfo]) -> None
 
 
 # Checking and Validation Functions
+# pylint: disable=too-many-locals
```

Comment:
pylint was up to something. By splitting this into 5 methods based on your numbered comments you would probably also indirectly avoid this error.